control-plane/api-gateway/translation/k8s_cache_translation.go

```diff
@@ -0,0 +1,131 @@
+package translation
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/hashicorp/consul/api"
+)
+
+type TranslatorFn func(api.ConfigEntry) []types.NamespacedName
+
+type secretTransfomer func(context.Context) func(client.Object) []reconcile.Request
+
+type resourceGetter interface {
+	Get(api.ResourceReference) api.ConfigEntry
+}
+
+// ConsulToNSNTranslator handles translating consul config entries to k8s namespaced names
+type ConsulToNSNTranslator struct {
+	cache resourceGetter
+}
+
+// NewConsulToNSNTranslator creates an instance of the ConsulToNSNTranslator
+func NewConsulToNSNTranslator(cache resourceGetter) ConsulToNSNTranslator {
+	return ConsulToNSNTranslator{cache: cache}
+}
+
+// TranslateConsulGateway creates a slice k8s types.NamespacedName from the meta fields of the api gateway config entry.
+func (c ConsulToNSNTranslator) TranslateConsulGateway(ctx context.Context) TranslatorFn {
+	return func(config api.ConfigEntry) []types.NamespacedName {
+		meta, ok := metaToK8sNamespacedName(config)
+		if !ok {
+			return nil
+		}
+
+		return []types.NamespacedName{meta}
+	}
+}
+
+// TranslateConsulHTTPRoute creates a slice of k8s types.NamespacedName from the meta fields of the http route parent refs.
+func (c ConsulToNSNTranslator) TranslateConsulHTTPRoute(ctx context.Context) TranslatorFn {
+	return func(config api.ConfigEntry) []types.NamespacedName {
+		route, ok := config.(*api.HTTPRouteConfigEntry)
+		if !ok {
+			return nil
+		}
+
+		return consulRefsToNSN(c.cache, route.Parents)
+	}
+}
+
+// TranslateConsulTCPRoute creates a slice of k8s types.NamespacedName from the meta fields of the tcp route parent refs.
+func (c ConsulToNSNTranslator) TranslateConsulTCPRoute(ctx context.Context) TranslatorFn {
+	return func(config api.ConfigEntry) []types.NamespacedName {
+		route, ok := config.(*api.TCPRouteConfigEntry)
+		if !ok {
+			return nil
+		}
+
+		return consulRefsToNSN(c.cache, route.Parents)
+	}
+}
```

Comment:
Same here as above, these seems to build functions rather than executing the translation.